common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -3757,13 +3761,34 @@ object CelebornConf extends Logging {
       .booleanConf
       .createWithDefault(true)
 
+  val CLIENT_PUSH_DYNAMIC_WRITE_MODE_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.client.spark.push.dynamicWriteMode.enabled")
+      .categories("client")
+      .doc("Whether to dynamically switch push write mode based on conditions.If true, " +
+        s"shuffle mode will be only determined by partition count")
+      .version("0.5.0")
+      .booleanConf
+      .createWithDefault(false)
+
+  val CLIENT_PUSH_DYNAMIC_WRITE_MODE_PARTITION_NUM_THRESHOLD: ConfigEntry[Int] =
+    buildConf("celeborn.client.spark.push.dynamicWriteMode.partitionNum.threshold")
+      .categories("client")
+      .doc("Threshold of shuffle partition count for push writer mode by partition count." +
+        s"If ${CLIENT_PUSH_DYNAMIC_WRITE_MODE_ENABLED.key} is true, Celeborn will use sort-based " +
+        s"shuffle writer when partition count is greater than this value," +
+        s"and used hash-based shuffle writer otherwise.")
```

Comment:
I edit it from web browser, you may need to break it into lines locally
```suggestion
      .doc("Threshold of shuffle partition number for dynamically switching push writer mode. When the shuffle partition number is greater than this value, use the sort-based shuffle writer for memory efficiency; otherwise use the hash-based shuffle writer for speed. This configuration only takes effect when ${CLIENT_PUSH_DYNAMIC_WRITE_MODE_ENABLED.key} is true.")
```

Reply:
Done